aws/data_aws_unity_catalog_policy.go

```diff
@@ -0,0 +1,102 @@
+package aws
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"regexp"
+)
+
+func DataAwsUnityCatalogPolicy() *schema.Resource {
```

Comment:
100-line & nested methods are not easy to review or to maintain. Let's un-inline the schema definition and ReadContext function as private functions within this file.